src/main/java/com/google/api/codegen/grpcmetadatagen/java/JavaPackageMetadataTransformer.java

```diff
@@ -17,30 +17,42 @@
 import com.google.api.codegen.TargetLanguage;
 import com.google.api.codegen.config.ApiModel;
 import com.google.api.codegen.config.PackageMetadataConfig;
-import com.google.api.codegen.grpcmetadatagen.GenerationLayer;
+import com.google.api.codegen.grpcmetadatagen.ArtifactType;
 import com.google.api.codegen.transformer.PackageMetadataTransformer;
 import com.google.api.codegen.transformer.java.JavaPackageMetadataNamer;
 import com.google.api.codegen.viewmodel.metadata.PackageDependencyView;
 import com.google.api.codegen.viewmodel.metadata.PackageMetadataView;
+import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.ImmutableSet;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 
 /** Responsible for producing package meta-data related views for Java */
-public abstract class JavaPackageMetadataTransformer {
+public class JavaPackageMetadataTransformer {
   // TODO determine if an API uses resource names from GAPIC config
   // https://github.com/googleapis/toolkit/issues/1668
   private ImmutableSet<String> SERVICES_WITH_NO_RESOURCE_NAMES =
       ImmutableSet.of("common-protos", "language", "speech", "video-intelligence", "vision");
 
   private final PackageMetadataTransformer metadataTransformer = new PackageMetadataTransformer();
+  private final Map<String, String> snippetsOutput;
+  private final ArtifactType artifactType;
 
-  protected abstract Map<String, String> getSnippetsOutput();
+  public JavaPackageMetadataTransformer(
+      Map<String, String> snippetsOutput, ArtifactType artifactType) {
+    this.snippetsOutput = ImmutableMap.copyOf(snippetsOutput);
+    this.artifactType = artifactType;
+  }
+
+  protected Map<String, String> getSnippetsOutput() {
```

Comment:
Can we make this either private or public? We aren't expecting subclasses any more.

Reply:
There is still one subclass remaining: `JavaGapicMetadataTransformer`. But it makes sence, `JavaPackageMetadataTransformer` transforms for proto and grpc `build.gradle`, `JavaGapicMetadataTransformer` transforms for gapic.

Reply:
But does this need to remain `protected`?

Reply:
It does not, but when I make it private, IntelliJ highlights it and says it can be protected. I still can make it public, of course, do you feel strong about it?

Reply:
Btw, `JavaGapicMetadataTransformer` will go away in my next cleanup of toolkit (to remove sample app stuff). So, eventually this method can become private.